vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -64,19 +64,18 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
             }
           });
 
-          ItemCache.prototype.getCacheAndIndexByKey = tryCatchWrapper(function (key) {
+          ItemCache.prototype.getCacheByKey = tryCatchWrapper(function (key) {
+            // Start looking in this cache
             for (let index in this.items) {
```

Comment:
**idea:** This could be further optimized by iterating through `itemkeyCaches`, which contains only the items that are expanded.

Reply:
I was actually planning to remove `itemkeyCaches` after this PR, because they don't have any purpose at the moment.

Reply:
Your idea would indeed make them relevant, it's a trade-off between optimizing this loop and reducing complexity / confusion in the connector. I kind of prefer the second option 🙂.

Reply:
Ah, that would be great.